fish

```diff
@@ -0,0 +1,33 @@
+# All pitfalls are taken from https://fishshell.com/docs/current/faq.html
+
+# Creating a variable:
+set -x variable value # typically set -gx key value
+
+# Removing a variable:
+set -e variable
+
+# Checking whether variable is defined:
+if set -q variable
+    some_command
+end
+
+# Checking whether variable is not empty:
+if string length -q -- $variable
```

Comment:
FWIW, many alternatives including `string match -rq . -- $variable` and `test -z "$variable"` (quotes required for the latter).

Reply:
What option is preferred by default?